### ISSUE TITLE
DCMTK sanity check fails due to missing OpenSSL library

### ISSUE THREAD

**Issue by bedroge** [easybuilders/easybuild-easyconfigs#14455]
My `DCMTK-3.6.6-GCCcore-10.3.0.eb` installation on CentOS 7 (with OpenSSL 1.0.2k) failed during the sanity check:
```
== 2021-11-29 08:27:31,748 build_log.py:169 ERROR EasyBuild crashed with an error (at easybuild/base/exceptions.py:124 in __init__): Sanity check failed: sanity check command dcmtls_tests | grep " 0 failed" exited with code 1 (output: dcmtls_tests: error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory
```

It looks like the build picked up the library from the `OpenSSL/1.1` module, which is not listed as a dependency, as the `CMakeCache.txt` also shows:
```
//Path to a library.
OPENSSL_CRYPTO_LIBRARY:FILEPATH=/software/software/OpenSSL/1.1/lib/libcrypto.so

//Path to a file.
OPENSSL_INCLUDE_DIR:PATH=/software/software/OpenSSL/1.1/include

//Path to a library.
pkgcfg_lib__OPENSSL_crypto:FILEPATH=/software/software/OpenSSL/1.1/lib/libcrypto.so

//Path to a library.
pkgcfg_lib__OPENSSL_ssl:FILEPATH=/software/software/OpenSSL/1.1/lib/libssl.so
```

And from the output of the configure step:
```
-- Performing Test OPENSSL_VERSION_CHECK
-- Performing Test OPENSSL_VERSION_CHECK - Success
-- Info: DCMTK OPENSSL support will be enabled
```

Also see:
https://github.com/DCMTK/dcmtk/blob/DCMTK-3.6.6/CMake/3rdparty.cmake#L60

I think this also applies to older versions, as I also see this is the same file for `3.6.3`:
https://github.com/DCMTK/dcmtk/blob/DCMTK-3.6.3/CMake/3rdparty.cmake#L19

But I'll give that version a try as well, and open a PR to address the issue for all relevant versions.

**Comment by bedroge** [comment]
Apparently, the other/older versions are not really affected; though they do need OpenSSL, they pick up the library from the host. It's actually the newer CMake that's pulling in the OpenSSL/1.1 module for th DCMTK 3.6.6.

In order to fix it, DCMTK 3.6.6 needs an explicit dependency on OpenSSL 1.1. For the older versions, I guess an OS dependency should be sufficient?

**Comment by bedroge** [comment]
The fix in #14456 was never merged, but it was fixed in a newer easyconfig (https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/d/DCMTK/DCMTK-3.6.7-GCCcore-11.3.0.eb), so I guess this can be closed now.
